presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -320,7 +325,9 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
             return true;
         }
-
+        if (!canAccessSchema(context, tableName.getSchemaName(), AccessMode.READ_ONLY)) {
```

Comment:
schema access mode should be calculated from `requiredPrivileges`. If user needs INSERT to a table but has only READ_ONLY to schema then I don't think it should be allowed.

Reply:
sure, AccessMode on schema should propagate to table control